src/dso_api/dynamic_api/serializers.py

```diff
@@ -388,14 +398,19 @@ def get_view_name(model: Type[DynamicModel], suffix: str):
 
 
 @lru_cache()
-def serializer_factory(
+def serializer_factory(  # noqa: C901
     model: Type[DynamicModel], depth: int, flat=None, links_serializer=False
 ) -> Type[DynamicSerializer]:
     """Generate the DRF serializer class for a specific dataset model.
     It can generate the serializer for both the body as well as for the _links field,
     depending on links_serializer being False or True."""
+    if isinstance(model, str):
+        raise ImproperlyConfigured(f"Model {model} could not be resolved.")
 
-    if links_serializer:
```

Comment:
Just a brainfart here: maybe it's time to actually split factory into 3 different ones: 
 - `serializer_factory`
 - `flat_serializer_factory`
 - `links_serializer_factory`
Otherwise single factory becomes too complex and prone to be broken by minor changes.

Reply:
great idea! Makes the code much cleaner too.